src/components/DialogContent.tsx

```diff
@@ -17,16 +17,42 @@ limitations under the License.
 import * as React from "react";
 import { ModuleApi } from "../ModuleApi";
 import { PlainSubstitution } from "../types/translations";
+import { ModuleUiDialogOptions } from "../types/ModuleUiDialogOptions";
 
+/** React properties for dialog content implementations based on {@link DialogContent} */
 export interface DialogProps {
+    /**
+     * A reference to the active Module API.
+     */
     moduleApi: ModuleApi;
+
+    /**
+     * Callback to update the dialog options.
+     * 
+     * Dialog content implementations can call this to update any of the options that were
+     * originally set via {@link ModuleApi.openDialog}.
+     * 
+     * @param options - The updates that should be applied to the dialog options. Any properties
+     * not set in the {@link options} are left unchanged.
+     */
+    setOptions(options: Partial<ModuleUiDialogOptions>): void;
+
+    /**
+     * Cancel the dialog programmatically.
+     */
+    cancel(): void;
 }
 
+/** State of {@link DialogContent} */
 export interface DialogState {
     busy: boolean;
     error?: string;
 }
 
+/**
+ * A base class for the content of a Dialog. This can be passed to the Module API with
+ * {@link ModuleApi.openDialog} to define the content inside the dialog.
```

Comment:
we can't pass this class directly to `openDialog`, right? Rather, it's what the `body` returns?

```suggestion
 * Base class for the content of a Dialog.
 * 
 * The `body` callback passed to {@link ModuleApi.openDialog} should return an instance of a
 * class based on this.
```

Reply:
Yes that's correct and definitely clearer 👍